File: app/src/main/java/com/mandarin/bcu/androidutil/io/drive/DriveUtil.java
package com.mandarin.bcu.androidutil.io.drive;

import android.util.Log;

import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
import com.google.api.client.http.FileContent;
import com.google.api.client.http.HttpTransport;
import com.google.api.client.http.javanet.NetHttpTransport;
import com.google.api.client.json.JsonFactory;
import com.google.api.client.json.jackson2.JacksonFactory;
import com.google.api.services.drive.Drive;
import com.google.api.services.drive.DriveScopes;
import com.google.api.services.drive.model.File;
import com.mandarin.bcu.androidutil.StaticStore;

import java.io.IOException;
import java.io.InputStream;
import java.util.Collections;

public class DriveUtil {
    public static boolean Upload(java.io.File file, InputStream in) throws IOException {
        GoogleCredential credential = GoogleCredential.fromStream(in).createScoped(Collections.singletonList(DriveScopes.DRIVE));

        HttpTransport transport = new NetHttpTransport();
        JsonFactory factory = JacksonFactory.getDefaultInstance();
        Drive service = new com.google.api.services.drive.Drive.Builder(transport,factory,credential).setApplicationName("BCU").build();

        Log.i("uploadDrive","-*****- Uploading Log File to Drive -> File Name : "+file.getName()+" | File Size : "+file.length()+" -*****-");

        File target = new File();

        FileContent con = new FileContent("text/txt",file);

        target.setName(file.getName()).setMimeType(con.getType())
                .setParents(Collections.singletonList(StaticStore.ERR_FILE_ID));

        service.files().create(target,con).execute();

        return true;
    }
}

File: app/src/main/java/com/mandarin/bcu/androidutil/io/drive/DriveUtil.kt
package com.mandarin.bcu.androidutil.io.drive

import android.util.Log
import com.google.api.client.http.FileContent
import com.google.api.client.http.HttpTransport
import com.google.api.client.http.javanet.NetHttpTransport
import com.google.api.client.json.JsonFactory
import com.google.api.client.json.gson.GsonFactory
import com.google.api.services.drive.Drive
import com.google.api.services.drive.DriveScopes
import com.google.auth.http.HttpCredentialsAdapter
import com.google.auth.oauth2.GoogleCredentials
import com.mandarin.bcu.androidutil.StaticStore
import java.io.File
import java.io.IOException
import java.io.InputStream

object DriveUtil {
    @Throws(IOException::class)
    fun upload(file: File, stream: InputStream?): Boolean {
        val credential = GoogleCredentials.fromStream(stream).createScoped(listOf(DriveScopes.DRIVE))

        val request = HttpCredentialsAdapter(credential)
        val transport: HttpTransport = NetHttpTransport()
        val factory: JsonFactory = GsonFactory.getDefaultInstance()
        val service = Drive.Builder(transport, factory, request).setApplicationName("BCU").build()

        Log.i("uploadDrive", "-*****- Uploading Log File to Drive -> File Name : " + file.name + " | File Size : " + file.length() + " -*****-")

        val target = com.google.api.services.drive.model.File()
        val con = FileContent("text/txt", file)

        target.setName(file.name).setMimeType(con.type).parents = listOf(StaticStore.ERR_FILE_ID)

        service.files().create(target, con).execute()
        
        return true
    }
}